File: Java - Microservices with Spring Boot, Spring Cloud/Glide-Smart-Bike-Rental-System-main/glide-frontend/src/services/reservationService.js
import api from "./api";

export const reserveBike = async (bikeId) => {

    const response = await api.post(`/api/reservations/${bikeId}`);

    return response.data;

};

export const getMyReservations = async () => {

    const response = await api.get("/api/reservations/my");

    return response.data;

};

export const cancelReservation = async (reservationId) => {

    const response = await api.delete(
        `/api/reservations/${reservationId}`
    );

    return response.data;

};

File: Java - Microservices with Spring Boot, Spring Cloud/Glide-Smart-Bike-Rental-System-main/glide-backend/src/main/java/com/glide/service/ReservationService.java
package com.glide.service;

import com.glide.dto.ReservationResponse;

import java.util.List;

public interface ReservationService {

    ReservationResponse reserveBike(Long bikeId);

    List<ReservationResponse> getMyReservations();

    void cancelReservation(Long reservationId);

}